soap/src/java/com/zimbra/soap/mail/type/NewFolderSpec.java

```diff
@@ -95,7 +95,7 @@
      * @zm-api-field-tag parent-folder-id
      * @zm-api-field-description Parent folder ID
      */
-    @XmlAttribute(name=MailConstants.A_FOLDER /* l */, required=false)
+    @XmlAttribute(name=MailConstants.A_FOLDER /* l */, required=true)
```

Comment:
changing the soap behaviour, please check it doesn't break anything.

Reply:
yes, this was already a mandatory attribute. It was throwing validation error if you don't pass this "l" attribute. I was also confused because of this false, so corrected it to true.

Reply:
Since this is a bug in the soap api, you should probably fix this on a separate ticket and open a PR to develop, otherwise it may be a while before this gets merged in.

Reply:
a separate ticket is raised for this ZCS-5770, but though of fixing this along with this.